### PR TITLE
#3373 IEC WT models miscellaneous errors

### PR DESCRIPTION
**Pull request by louisg-rte**
**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [ ] unit tests and non-regression tests were added (new model, new feature and bug fix)
- [ ] main documentation was updated (update of input/output file, 3rd party, model, repository organization, solver)
- [ ] example documentations were updated (new example in examples folder)
- [ ] the corresponding milestone was added in the ticket and in this PR
- [ ] if this PR modifies the parameters or inputs/outputs of a model/solver: the corresponding xsl was added in util/xsl and platform DB were updated
- [ ] if this PR modifies a dictionary: the corresponding french dictionary was updated

### CODE REVIEW

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/IEC/IEC61400/Parameters/QControlParameters2015.mo [hunk 1]
```diff
@@ -0,0 +1,45 @@
+within Dynawo.Electrical.Controls.IEC.IEC61400.Parameters;
+
+/*
+* Copyright (c) 2022, RTE (http://www.rte-france.com)
```
Comment (joyelfeghali) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/IEC/IEC61400/Parameters/QControlParameters2015.mo:4]: 2024

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/IEC/Parameters/QControlParameters2015.mo
Comment (joyelfeghali) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/IEC/Parameters/QControlParameters2015.mo:1]: Why did you add the parameters also here IEC/Parameters/QControlParameters2015.mo and IEC/Parameters/QControlParameters2020.mo ?

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/IEC/IEC61400/BaseClasses/BaseWPPQControl.mo [hunk 2]
```diff
@@ -49,7 +48,7 @@ model BaseWPPQControl "Reactive power control base module for wind power plants
 
   Modelica.Blocks.Math.Feedback feedback annotation(
     Placement(visible = true, transformation(origin = {-220, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
-  Modelica.Blocks.Tables.CombiTable1Ds combiTable1Ds2(extrapolation = Modelica.Blocks.Types.Extrapolation.HoldLastPoint, table = TableQwpUErr) annotation(
+  Modelica.Blocks.Tables.CombiTable1Ds combiTable1Ds2(extrapolation = Modelica.Blocks.Types.Extrapolation.HoldLastPoint)  annotation(
```
Comment (joyelfeghali) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/IEC/IEC61400/BaseClasses/BaseWPPQControl.mo:51]: Why remove the table's name?  We have the same one for 2015 and 2020 no?

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/IEC/IEC61400/Parameters/package.order [hunk 3]
```diff
@@ -1,5 +1,6 @@
 CurrentLimitParameters
 GridProtectionParameters
 PControlParameters
-QControlParameters
 QLimitParameters
+QControlParameters2015
+QControlParameters2020
```
Comment (joyelfeghali) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/IEC/IEC61400/Parameters/package.order:6]: alphabetic order

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/IEC/IEC61400/BaseControls/Auxiliaries/UfMeasurement.mo [hunk 4]
```diff
@@ -62,23 +72,31 @@ equation
...
+  connect(complexToPolar.phi, firstOrder2.u) annotation(
+    Line(points = {{-60, 40}, {-56, 40}, {-56, 24}, {-46, 24}}, color = {0, 0, 127}));
+  connect(firstOrder2.y, switch1.u1) annotation(
+    Line(points = {{-22, 24}, {-2, 24}}, color = {0, 0, 127}));
+  connect(booleanConstant.y, switch1.u2) annotation(
+    Line(points = {{-22, -18}, {-12, -18}, {-12, 32}, {-2, 32}}, color = {255, 0, 255}));
   annotation(
     Diagram(coordinateSystem(extent = {{-100, -100}, {150, 100}})),
     Icon(coordinateSystem(extent = {{-100, -900}, {100, 900}}), graphics = {Rectangle(fillColor = {255, 255, 255}, fillPattern = FillPattern.Solid, extent = {{-100, 900}, {100, -900}}), Text(origin = {-1, 24}, rotation = -90, extent = {{-441, 82}, {441, -82}}, textString = "Uf measurement")}));
+
```
Comment (joyelfeghali) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/IEC/IEC61400/BaseControls/Auxiliaries/UfMeasurement.mo:101]: line skip before annotation instead of here

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Wind/IEC/BaseClasses/BaseWPP.mo [hunk 5]
```diff
@@ -14,8 +14,9 @@ within Dynawo.Electrical.Wind.IEC.BaseClasses;
 
 model BaseWPP "Base model for Wind Power Plants from IEC 61400-27-1 standard"
   extends Dynawo.Electrical.Controls.IEC.IEC61400.Parameters.GridProtectionParameters;
+  extends Dynawo.Electrical.Controls.IEC.IEC61400.Parameters.CurrentLimitParameters;
+  extends Dynawo.Electrical.Controls.IEC.IEC61400.Parameters.QLimitParameters;
```
Comment (joyelfeghali) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Wind/IEC/BaseClasses/BaseWPP.mo:18]: alphabetic order

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Wind/IEC/BaseClasses/BaseWT4.mo [hunk 6]
```diff
@@ -14,6 +14,8 @@ within Dynawo.Electrical.Wind.IEC.BaseClasses;
 
 partial model BaseWT4 "Base model for Wind Turbine Type 4 from IEC 61400-27-1 standard"
   extends Dynawo.Electrical.Controls.IEC.IEC61400.Parameters.GridProtectionParameters;
+  extends Dynawo.Electrical.Controls.IEC.IEC61400.Parameters.CurrentLimitParameters;
+  extends Dynawo.Electrical.Controls.IEC.IEC61400.Parameters.QLimitParameters;
```
Comment (joyelfeghali) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Wind/IEC/BaseClasses/BaseWT4.mo:18]: same

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Wind/IEC/WPP/WPP4ACurrentSource2015.mo [hunk 7]
```diff
@@ -14,12 +14,17 @@ within Dynawo.Electrical.Wind.IEC.WPP;
...
+  extends Dynawo.Electrical.Controls.IEC.IEC61400.Parameters.QControlParameters2015;
 
   //Uf measurement parameters
   parameter Types.AngularVelocityPu DfMaxPu "Maximum frequency ramp rate in pu/s (base omegaNom)" annotation(
     Dialog(tab = "UfMeasurement"));
+  parameter Boolean Mzc "Zero crossing measurement mode (true = 1 if the wind turbine protection system uses zero crossings to detect the frequency - otherwise false = 0)" annotation(
+    Dialog(tab = "UfMeasurement"));
   parameter Types.Time tfFilt "Filter time constant for frequency measurement in s" annotation(
     Dialog(tab = "UfMeasurement"));
+  parameter Types.Time tphiFilt "Filter time constant for voltage angle measurement in s" annotation(
```
Comment (joyelfeghali) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Wind/IEC/WPP/WPP4ACurrentSource2015.mo:26]: could you update the Dynawo.Examples.Wind.IEC.Neplan.WPP4ACurrentSource2015 example, by putting values for Mzc and tphiFilt ?

**Review by joyelfeghali — CHANGES_REQUESTED** [review]

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/IEC/Parameters/QControlParameters2015.mo
Comment (louisg-rte) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/IEC/Parameters/QControlParameters2015.mo:1]: I've done that because the combitables TableQwpMaxPwpFiltCom and TableQwpMinPwpFiltCom are not defined in the 2015 version

Comment (joyelfeghali) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/IEC/Parameters/QControlParameters2015.mo:1]: Yes I understand but you added the parameters file twice, I think it was a mistake? Because you don't use them also they are not seen on OpenModelica because you didn't add them to a package order or CMakeList

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/IEC/IEC61400/BaseClasses/BaseWPPQControl.mo (hunk 2 above)
Comment (louisg-rte) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/IEC/IEC61400/BaseClasses/BaseWPPQControl.mo:51]: Considering that the tables are not declared in this base model BaseWPPQControl.mo (because they are different between 2015 and 2020 versions) I don't think it is possible to call them here. So I've done it WPPQControl2015 and WPPQControl2020

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Wind/IEC/WPP/WPP4ACurrentSource2015.mo (hunk 7 above)
Comment (louisg-rte) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Wind/IEC/WPP/WPP4ACurrentSource2015.mo:26]: I think this is already done for all the concerned examples

**Review by joyelfeghali — APPROVED** [review]
Could you squash the commits please?

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/IEC/IEC61400/Parameters/QControlParameters2020.mo [hunk 8]
```diff
@@ -12,7 +12,7 @@ within Dynawo.Electrical.Controls.IEC.IEC61400.Parameters;
 * This file is part of Dynawo, an hybrid C++/Modelica open source suite of simulation tools for power systems.
 */
 
-record QControlParameters
+record QControlParameters2020
```
Comment (joyelfeghali) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/IEC/IEC61400/Parameters/QControlParameters2020.mo:15]: Also, could you add this change to the updateXML file? you can put it in update1.7.5.py

Comment (louisg-rte) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/IEC/IEC61400/Parameters/QControlParameters2020.mo:15]: I've put it in update1.7.0.6.py considering what was done for the previous tickets is it ok ?

**Review by joyelfeghali — APPROVED** [review]
